flink-connector-redis/src/test/java/org/apache/flink/streaming/connectors/redis/RedisSinkITCase.java

```diff
@@ -127,6 +146,24 @@ public void testRedisHashDataType() throws Exception {
         env.execute("Test Redis Hash Data Type");
 
         assertEquals(NUM_ELEMENTS, jedis.hlen(REDIS_ADDITIONAL_KEY));
+        Thread.sleep(REDIS_TTL * 1000);
+        assertEquals(true, jedis.exists(REDIS_ADDITIONAL_KEY));
+
+        jedis.del(REDIS_ADDITIONAL_KEY);
+    }
+
+    @Test
+    public void testRedisHashDataTypeWithTTL() throws Exception {
```

Comment:
You're not testing the case where the TTL comes from the value (the mapper extracts it)